src/jdk.jartool/share/classes/sun/tools/jar/Main.java

```diff
@@ -1743,12 +1749,23 @@ private File createTemporaryFile(String tmpbase, String suffix) {
 
     /**
      * Associates a module descriptor's zip entry name along with its
-     * bytes and an optional URI. Used when describing modules.
+     * bytes and an optional URI.
      */
     interface ModuleInfoEntry {
        String name();
        Optional<String> uriString();
        InputStream bytes() throws IOException;
+        /**
+         * @return Returns the last modified time of the module descriptor.
+         * Returns null if the last modified time is unknown or cannot be
+         * determined.
+         */
+       FileTime getLastModifiedTime();
+       default byte[] readAllBytes() throws IOException {
+            try (InputStream is = bytes()) {
+                return is.readAllBytes();
+            }
+       }
```

Comment:
I think the comment would be clear if it says "the last modified time of the module-info.class".
Also would you mind changing ModuleInfoEntry to use 4-space rather than 2-space indentation?